app/models/types/patterns/token_property_mapper.rb

```diff
@@ -31,7 +31,11 @@
 module Types
   module Patterns
     class TokenPropertyMapper
-      DEFAULT_FUNCTION = ->(key, context) { context.public_send(key.to_sym) }.curry
+      DEFAULT_FUNCTION = ->(key, context) do
+        return nil if context.nil?
```

Comment:
I think we should raise an exception here. This should never happen.

Reply:
Huh? of course this happens. If you have a wp without a parent. ;)